include/rtc/rtp.hpp

```diff
@@ -335,6 +338,39 @@ struct RTCP_SDES_CHUNK {
         return reinterpret_cast<RTCP_SDES_ITEM *>(base);
     }
 
+    long safelyCountChunkSize(unsigned int maxChunkSize) {
+        if (maxChunkSize < RTCP_SDES_CHUNK::size({})) {
+            // chunk is truncated
+            return -1;
+        } else {
+            unsigned int size = sizeof(SSRC);
+            unsigned int i = 0;
+            // We can always access first 4 bytes of first item (in case of no items there will be 4 null bytes)
+            auto item = getItem(i);
+            std::vector<uint8_t> textsLength{};
+            while (item->type != 0) {
+                if (size + RTCP_SDES_ITEM::size(0) > maxChunkSize) {
+                    // item is too short
+                    return -1;
+                }
+                auto itemLength = item->length();
+                if (size + RTCP_SDES_ITEM::size(itemLength) >= maxChunkSize) {
+                    // item is too large (it can't be equal to chunk size because after item there must be 1-4 null bytes as padding)
+                    return -1;
+                }
+                textsLength.push_back(itemLength);
+                // safely to access next item
+                item = getItem(++i);
+            }
+            auto realSize = RTCP_SDES_CHUNK::size(textsLength);
+            if (realSize > maxChunkSize) {
+                // Chunk is too large
+                return -1;
+            }
+            return realSize;
+        }
+    }
+
     [[nodiscard]] static unsigned int size(std::vector<uint8_t> textLengths) {
```

Comment:
This should probably be a const reference to avoid the copy.

Reply:
Fixed.